Android widget images not loaded: No implementation found for `boolean org.xbmc.kodi.XBMCFile._open(java.lang.String)`

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->

1. Install Kodi onto Android (in my case it's Android TV)
2. Add its channel widget into the home screen (in my case I can choose between "Movies" and "Recommendations")

The images of the recommendation cards are not visible on the screen, they are replaced by a grey question mark.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->

The card images show the recommended movie arts.

### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:

The launcher tries to show multiple cards at once.

Here are the ADB log messages from the beginning:
```
03-15 09:06:42.127  7982  7982 W Glide   : Load failed for [null] with dimensions [0x0]
03-15 09:06:42.127  7982  7982 W Glide   : class v3.a0: Received null model
03-15 09:06:42.128  7982  7982 W Glide   : Load failed for [null] with dimensions [0x0]
03-15 09:06:42.128  7982  7982 W Glide   : class v3.a0: Received null model
03-15 09:06:42.132 20252 22725 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.132 20252 22725 E Kodifile: Open: Not available
03-15 09:06:42.133  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.134 20252 22726 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.134 20252 22726 E Kodifile: Open: Not available
03-15 09:06:42.135  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.135  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.135  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.135 20252 22727 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.135 20252 22727 E Kodifile: Open: Not available
03-15 09:06:42.135  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.136  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.136  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.136  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.136 20252 22728 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.136 20252 22728 E Kodifile: Open: Not available
03-15 09:06:42.137  7982  8169 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.139  7982  8169 I chatty  : uid=10116(com.spocky.projengmenu) glide-source-th identical 4 lines
03-15 09:06:42.139  7982  8169 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.139  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.140  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.140  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.141  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.142  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.142 20252 22730 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.142 20252 22730 E Kodifile: Open: Not available
03-15 09:06:42.142  7982  8173 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.143  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.143  7982  8172 I chatty  : uid=10116(com.spocky.projengmenu) glide-source-th identical 2 lines
03-15 09:06:42.144  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.145 20252 22729 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.145 20252 22729 E Kodifile: Open: Not available
03-15 09:06:42.146 20252 22733 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.146 20252 22733 E Kodifile: Open: Not available
03-15 09:06:42.146 20252 22731 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.146 20252 22731 E Kodifile: Open: Not available
03-15 09:06:42.147 20252 22732 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.147 20252 22732 E Kodifile: Open: Not available
03-15 09:06:42.147  7982  8174 E MediaMetadataRetrieverJNI: negative length (-1)
03-15 09:06:42.149  7982  8173 E MediaMetadataRetrieverJNI: negative length (-1)
03-15 09:06:42.149  7982  8172 E MediaMetadataRetrieverJNI: negative length (-1)
03-15 09:06:42.151  7982  8169 E MediaMetadataRetrieverJNI: negative length (-1)
03-15 09:06:42.151 20252 22734 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.151 20252 22734 E Kodifile: Open: Not available
03-15 09:06:42.152 20252 22735 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.152 20252 22735 E Kodifile: Open: Not available
03-15 09:06:42.153 20252 22738 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.153 20252 22738 E Kodifile: Open: Not available
03-15 09:06:42.153 20252 22736 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.153 20252 22737 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.153  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.153 20252 22737 E Kodifile: Open: Not available
03-15 09:06:42.154  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.154  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.154  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.154  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.154  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8172 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.155  7982  8174 D skia    : --- Failed to create image decoder with message 'unimplemented'
03-15 09:06:42.156 20252 22736 E Kodifile: Open: Not available
03-15 09:06:42.157 20252 22739 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.157 20252 22739 E Kodifile: Open: Not available
03-15 09:06:42.158 20252 22740 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.158 20252 22740 E Kodifile: Open: Not available
03-15 09:06:42.160 20252 22741 E org.xbmc.kodi: No implementation found for boolean org.xbmc.kodi.XBMCFile._open(java.lang.String) (tried Java_org_xbmc_kodi_XBMCFile__1open and Java_org_xbmc_kodi_XBMCFile__1open__Ljava_lang_String_2)
03-15 09:06:42.160 20252 22741 E Kodifile: Open: Not available
```

And here is an exception which is repeated for each of the cards. I paste only one here.

The movies are stored on local network (I use the Jellyfin plugin to retrieve them), that's why you see a network location in the logs. These logs are probably from the the launcher itself which wants to load the Kodi-provided images. Can be useful for your debugging but feel free to ignore it.

<details>
 <summary>Glide exception</summary>

  ```
03-15 09:06:42.161  7982  7982 W Glide   : Load failed for [content://org.xbmc.kodi.file#image%3A%2F%2Fhttp%253a%252f%252f192.168.XXX.XXX%253a8096%252fItems%252f036d6a0bc458583331de441XXXXXXXXX%252fImages%252fPrimary%252f0%253fFormat%253doriginal%2526Tag%253db2092bfc51b2956eb90bXXXXXXXXXXX%2F] with dimensions [105x158]
03-15 09:06:42.161  7982  7982 W Glide   : class v3.a0: Failed to load resource
03-15 09:06:42.161  7982  7982 W Glide   : There were 4 root causes:
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.lang.IllegalArgumentException(null)
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :   Cause (1 of 4): class v3.a0: Failed LoadPath{AutoCloseInputStream->Object->Drawable}, LOCAL
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (1 of 4): class v3.a0: Failed DecodePath{AutoCloseInputStream->Drawable->Drawable}
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (2 of 4): class v3.a0: Failed DecodePath{AutoCloseInputStream->c->Drawable}
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (3 of 4): class v3.a0: Failed DecodePath{AutoCloseInputStream->Bitmap->BitmapDrawable}
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (4 of 4): class v3.a0: Failed DecodePath{AutoCloseInputStream->BitmapDrawable->Drawable}
03-15 09:06:42.161  7982  7982 W Glide   :   Cause (2 of 4): class v3.a0: Failed LoadPath{ParcelFileDescriptorInner->Object->Drawable}, LOCAL
03-15 09:06:42.161  7982  7982 W Glide   : There were 3 root causes:
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (1 of 2): class v3.a0: Failed DecodePath{ParcelFileDescriptorInner->Bitmap->BitmapDrawable}
03-15 09:06:42.161  7982  7982 W Glide   : There were 2 root causes:
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :       Cause (1 of 2): class java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 W Glide   :       Cause (2 of 2): class java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (2 of 2): class v3.a0: Failed DecodePath{ParcelFileDescriptorInner->BitmapDrawable->Drawable}
03-15 09:06:42.161  7982  7982 W Glide   : There was 1 root cause:
03-15 09:06:42.161  7982  7982 W Glide   : java.io.IOException(android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek))
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :       Cause (1 of 1): class java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 W Glide   :   Cause (3 of 4): class v3.a0: Failed LoadPath{AssetFileDescriptor->Object->Drawable}, LOCAL
03-15 09:06:42.161  7982  7982 W Glide   : There was 1 root cause:
03-15 09:06:42.161  7982  7982 W Glide   : java.lang.IllegalArgumentException(null)
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (1 of 1): class v3.a0: Failed DecodePath{AssetFileDescriptor->Bitmap->BitmapDrawable}
03-15 09:06:42.161  7982  7982 W Glide   : There was 1 root cause:
03-15 09:06:42.161  7982  7982 W Glide   : java.lang.IllegalArgumentException(null)
03-15 09:06:42.161  7982  7982 W Glide   :  call GlideException#logRootCauses(String) for more detail
03-15 09:06:42.161  7982  7982 W Glide   :       Cause (1 of 1): class java.lang.IllegalArgumentException: 
03-15 09:06:42.161  7982  7982 W Glide   :   Cause (4 of 4): class v3.a0: Failed LoadPath{StringUri->Object->Drawable}, LOCAL
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (1 of 2): class v3.a0: Failed DecodePath{StringUri->Drawable->Drawable}
03-15 09:06:42.161  7982  7982 W Glide   :     Cause (2 of 2): class v3.a0: Failed DecodePath{StringUri->Bitmap->BitmapDrawable}
03-15 09:06:42.161  7982  7982 I Glide   : Root cause (1 of 4)
03-15 09:06:42.161  7982  7982 I Glide   : java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:17)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.b(Unknown Source:4)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.a(SourceFile:10)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.n.b(Unknown Source:16)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.n.a(SourceFile:28)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.c0.a(SourceFile:42)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.f(SourceFile:95)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.e(SourceFile:16)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.g(SourceFile:60)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.g(SourceFile:190)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.s(Unknown Source:48)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.run(Unknown Source:23)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-15 09:06:42.161  7982  7982 I Glide   :      at androidx.appcompat.widget.j.run(SourceFile:2230)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.lang.Thread.run(Thread.java:923)
03-15 09:06:42.161  7982  7982 I Glide   :      at y3.a.run(SourceFile:12)
03-15 09:06:42.161  7982  7982 I Glide   : Caused by: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.Linux.lseek(Native Method)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.BlockGuardOs.lseek(BlockGuardOs.java:223)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.161  7982  7982 I Glide   :      at android.system.Os.lseek(Os.java:357)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:10)
03-15 09:06:42.161  7982  7982 I Glide   :      ... 26 more
03-15 09:06:42.161  7982  7982 I Glide   : Root cause (2 of 4)
03-15 09:06:42.161  7982  7982 I Glide   : java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:17)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.b(Unknown Source:4)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.a(SourceFile:10)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.n.b(Unknown Source:16)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.n.a(SourceFile:28)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.c0.a(SourceFile:42)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.f(SourceFile:95)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.e(SourceFile:16)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.g(SourceFile:60)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.g(SourceFile:190)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.s(Unknown Source:48)
03-15 09:06:42.161  7982  7982 I Glide   :      at v3.m.run(Unknown Source:23)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-15 09:06:42.161  7982  7982 I Glide   :      at androidx.appcompat.widget.j.run(SourceFile:2230)
03-15 09:06:42.161  7982  7982 I Glide   :      at java.lang.Thread.run(Thread.java:923)
03-15 09:06:42.161  7982  7982 I Glide   :      at y3.a.run(SourceFile:12)
03-15 09:06:42.161  7982  7982 I Glide   : Caused by: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.Linux.lseek(Native Method)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.BlockGuardOs.lseek(BlockGuardOs.java:223)
03-15 09:06:42.161  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.161  7982  7982 I Glide   :      at android.system.Os.lseek(Os.java:357)
03-15 09:06:42.161  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:10)
03-15 09:06:42.161  7982  7982 I Glide   :      ... 26 more
03-15 09:06:42.162  7982  7982 I Glide   : Root cause (3 of 4)
03-15 09:06:42.162  7982  7982 I Glide   : java.io.IOException: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:17)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.b(Unknown Source:4)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.n.a(SourceFile:10)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.n.b(Unknown Source:16)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.n.a(SourceFile:28)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.c0.a(SourceFile:42)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.f(SourceFile:95)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.e(SourceFile:16)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.g(SourceFile:60)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.g(SourceFile:190)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.s(Unknown Source:48)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.run(Unknown Source:23)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-15 09:06:42.162  7982  7982 I Glide   :      at androidx.appcompat.widget.j.run(SourceFile:2230)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.lang.Thread.run(Thread.java:923)
03-15 09:06:42.162  7982  7982 I Glide   :      at y3.a.run(SourceFile:12)
03-15 09:06:42.162  7982  7982 I Glide   : Caused by: android.system.ErrnoException: lseek failed: ESPIPE (Illegal seek)
03-15 09:06:42.162  7982  7982 I Glide   :      at libcore.io.Linux.lseek(Native Method)
03-15 09:06:42.162  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.162  7982  7982 I Glide   :      at libcore.io.BlockGuardOs.lseek(BlockGuardOs.java:223)
03-15 09:06:42.162  7982  7982 I Glide   :      at libcore.io.ForwardingOs.lseek(ForwardingOs.java:150)
03-15 09:06:42.162  7982  7982 I Glide   :      at android.system.Os.lseek(Os.java:357)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.ParcelFileDescriptorRewinder$InternalRewinder.rewind(Unknown Source:10)
03-15 09:06:42.162  7982  7982 I Glide   :      ... 26 more
03-15 09:06:42.162  7982  7982 I Glide   : Root cause (4 of 4)
03-15 09:06:42.162  7982  7982 I Glide   : java.lang.IllegalArgumentException
03-15 09:06:42.162  7982  7982 I Glide   :      at android.media.MediaMetadataRetriever.setDataSource(Native Method)
03-15 09:06:42.162  7982  7982 I Glide   :      at m8.b.n(SourceFile:22)
03-15 09:06:42.162  7982  7982 I Glide   :      at c4.f0.b(SourceFile:88)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.n.b(Unknown Source:30)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.n.a(SourceFile:28)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.c0.a(SourceFile:42)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.f(SourceFile:95)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.e(SourceFile:16)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.g(SourceFile:60)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.g(SourceFile:190)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.g(SourceFile:190)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.c(Unknown Source:39)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.google.android.gms.internal.measurement.n3.o(SourceFile:72)
03-15 09:06:42.162  7982  7982 I Glide   :      at com.bumptech.glide.load.data.b.f(SourceFile:54)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.i0.b(SourceFile:153)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.r(SourceFile:26)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.s(Unknown Source:48)
03-15 09:06:42.162  7982  7982 I Glide   :      at v3.m.run(Unknown Source:23)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
03-15 09:06:42.162  7982  7982 I Glide   :      at androidx.appcompat.widget.j.run(SourceFile:2230)
03-15 09:06:42.162  7982  7982 I Glide   :      at java.lang.Thread.run(Thread.java:923)
03-15 09:06:42.162  7982  7982 I Glide   :      at y3.a.run(SourceFile:12)
  ```
</details>

### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [X] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android 11.0
 - Kodi version: 20.5



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/cmake/scripts/android/Install.cmake
+++ b/cmake/scripts/android/Install.cmake
@@ -48,6 +48,7 @@ set(package_files strings.xml
                   src/XBMCProperties.java
                   src/XBMCVideoView.java
                   src/XBMCFile.java
+                  src/XBMCTextureCache.java
                   src/XBMCURIUtils.java
                   src/channels/SyncChannelJobService.java
                   src/channels/SyncProgramsJobService.java

--- a/tools/android/packaging/xbmc/src/XBMCJsonRPC.java.in
+++ b/tools/android/packaging/xbmc/src/XBMCJsonRPC.java.in
@@ -24,15 +24,15 @@ import android.net.Uri;
 import android.provider.BaseColumns;
 import android.util.Log;
 
+import @APP_PACKAGE@.content.XBMCFileContentProvider;
 import @APP_PACKAGE@.model.Album;
+import @APP_PACKAGE@.model.File;
+import @APP_PACKAGE@.model.Media;
 import @APP_PACKAGE@.model.Movie;
 import @APP_PACKAGE@.model.MusicVideo;
 import @APP_PACKAGE@.model.Song;
 import @APP_PACKAGE@.model.TVEpisode;
 import @APP_PACKAGE@.model.TVShow;
-import @APP_PACKAGE@.content.XBMCFileContentProvider;
-import @APP_PACKAGE@.model.File;
-import @APP_PACKAGE@.model.Media;
 
 public class XBMCJsonRPC
 {
@@ -63,6 +63,7 @@ public class XBMCJsonRPC
 
   private String m_xbmc_web_url = "http://localhost:8080";
   private HashSet<Integer> mRecomendationIds = new HashSet<Integer>();
+  private XBMCTextureCache mTextureCache = null;
 
   private int MAX_RECOMMENDATIONS = 3;
 
@@ -126,6 +127,7 @@ public class XBMCJsonRPC
   {
     String jsonPort = XBMCProperties.getStringProperty("xbmc.jsonPort", "8080");
     m_xbmc_web_url = "http://localhost:" + jsonPort;
+    mTextureCache = new XBMCTextureCache();
   }
 
   public String request_string(String jsonRequest)
@@ -864,7 +866,7 @@ public class XBMCJsonRPC
       String poster = extractKeyFromArtMap(details, "poster");
       if (poster != null && !poster.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(poster).toString());
+        med.setCardImageUrl(getImageUrl(poster));
         med.setCardImageAspectRatio("2:3");
       }
       // fallback to thumb
@@ -873,7 +875,7 @@ public class XBMCJsonRPC
         poster = extractKeyFromArtMap(details, "thumb");
         if (poster != null && !poster.isEmpty())
         {
-          med.setCardImageUrl(XBMCFileContentProvider.buildUri(poster).toString());
+          med.setCardImageUrl(getImageUrl(poster));
           med.setCardImageAspectRatio("16:9");
         }
       }
@@ -881,7 +883,7 @@ public class XBMCJsonRPC
       String fanart = extractKeyFromArtMap(details, "fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
 
       med.setXbmcUrl("videodb://movies/titles/" + details.get("movieid").getAsString() + "?showinfo=true");
@@ -935,7 +937,7 @@ public class XBMCJsonRPC
       String poster = extractKeyFromArtMap(details, "poster");
       if (poster != null && !poster.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(poster).toString());
+        med.setCardImageUrl(getImageUrl(poster));
         med.setCardImageAspectRatio("2:3");
       }
       // fallback to thumb
@@ -944,7 +946,7 @@ public class XBMCJsonRPC
         poster = extractKeyFromArtMap(details, "thumb");
         if (poster != null && !poster.isEmpty())
         {
-          med.setCardImageUrl(XBMCFileContentProvider.buildUri(poster).toString());
+          med.setCardImageUrl(getImageUrl(poster));
           med.setCardImageAspectRatio("16:9");
         }
       }
@@ -952,7 +954,7 @@ public class XBMCJsonRPC
       String fanart = extractKeyFromArtMap(details, "fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
       med.setXbmcUrl("videodb://tvshows/titles/" + details.get("tvshowid").getAsInt() + "/");
       med.setCategory(Media.MEDIA_TYPE_TVSHOW);
@@ -982,14 +984,14 @@ public class XBMCJsonRPC
       String thumb = extractKeyFromArtMap(details, "thumb");
       if (thumb != null && !thumb.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+        med.setCardImageUrl(getImageUrl(thumb));
         med.setCardImageAspectRatio("16:9");
       }
       // fanart
       String fanart = extractKeyFromArtMap(details, "fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
 
       med.setXbmcUrl("videodb://tvshows/titles/" + details.get("tvshowid").getAsInt() + "/" + details.get("episodeid").getAsInt() + "?showinfo=true");
@@ -1029,14 +1031,14 @@ public class XBMCJsonRPC
       String thumb = extractKeyFromArtMap(details, "thumb");
       if (thumb != null && !thumb.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+        med.setCardImageUrl(getImageUrl(thumb));
         med.setCardImageAspectRatio("1:1");
       }
       // fanart
       String fanart = extractKeyFromArtMap(details, "fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
 
       med.setXbmcUrl("musicdb://albums/" + details.get("albumid").getAsString() + "/");
@@ -1064,7 +1066,7 @@ public class XBMCJsonRPC
       String thumb = extractKeyFromArtMap(details, "album.thumb");
       if (thumb != null && !thumb.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+        med.setCardImageUrl(getImageUrl(thumb));
         med.setCardImageAspectRatio("1:1");
       }
       // fallback to albumartist.thumb
@@ -1073,7 +1075,7 @@ public class XBMCJsonRPC
         thumb = extractKeyFromArtMap(details, "albumartist.thumb");
         if (thumb != null && !thumb.isEmpty())
         {
-          med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+          med.setCardImageUrl(getImageUrl(thumb));
           med.setCardImageAspectRatio("1:1");
         }
         else
@@ -1082,7 +1084,7 @@ public class XBMCJsonRPC
           thumb = extractKeyFromArtMap(details, "artist.thumb");
           if (thumb != null && !thumb.isEmpty())
           {
-            med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+            med.setCardImageUrl(getImageUrl(thumb));
             med.setCardImageAspectRatio("1:1");
           }
         }
@@ -1091,14 +1093,14 @@ public class XBMCJsonRPC
       String fanart = extractKeyFromArtMap(details, "albumartist.fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
       else
       {
         fanart = extractKeyFromArtMap(details, "artist.fanart");
         if (fanart != null && !fanart.isEmpty())
         {
-          med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+          med.setBackgroundImageUrl(getImageUrl(fanart));
         }
       }
 
@@ -1145,21 +1147,21 @@ public class XBMCJsonRPC
       String thumb = extractKeyFromArtMap(details, "thumb");
       if (thumb != null && !thumb.isEmpty())
       {
-        med.setCardImageUrl(XBMCFileContentProvider.buildUri(thumb).toString());
+        med.setCardImageUrl(getImageUrl(thumb));
         med.setCardImageAspectRatio("1:1");
       }
       // fanart
       String fanart = extractKeyFromArtMap(details, "fanart");
       if (fanart != null && !fanart.isEmpty())
       {
-        med.setBackgroundImageUrl(XBMCFileContentProvider.buildUri(fanart).toString());
+        med.setBackgroundImageUrl(getImageUrl(fanart));
       }
 
       med.setXbmcUrl("videodb://musicvideos/titles/" + details.get("musicvideoid").getAsInt());
 
       String url = getDownloadUrl(details.get("file").getAsString());
       if (url != null && !url.isEmpty())
-        med.setVideoUrl(XBMCFileContentProvider.buildUri(url).toString());
+        med.setVideoUrl(getImageUrl(url));
 
       med.setCategory(Media.MEDIA_TYPE_MUSICVIDEO);
     }
@@ -1436,6 +1438,25 @@ public class XBMCJsonRPC
       return null;
 
     return String.valueOf(rating / 2.0);
+  }
+
+  private String getImageUrl(String sUrl)
+  {
+    Log.d(TAG, "getImageUrl: sUrl = " + sUrl);
+    if (sUrl.startsWith("image://video@") || sUrl.startsWith("image://music@"))
+    {
+      Log.d(TAG, "getImageUrl: " + sUrl + " is not unwrapped");
+      return XBMCFileContentProvider.buildUri(sUrl).toString();
+    }
+
+    String sUnwrapImageUrl = this.mTextureCache.unwrapImageURL(sUrl);
+    Log.d(TAG, "getImageUrl: sUnwrapImageUrl = " + sUnwrapImageUrl);
+    if (!sUnwrapImageUrl.startsWith("http"))
+    {
+      return XBMCFileContentProvider.buildUri(sUrl).toString();
+    }
+
+    return sUnwrapImageUrl;
   }
 
 }

--- a/tools/android/packaging/xbmc/src/XBMCTextureCache.java.in
+++ b/tools/android/packaging/xbmc/src/XBMCTextureCache.java.in
@@ -1,0 +1,32 @@
+package @APP_PACKAGE@;
+
+import android.util.Log;
+
+/**
+ * Created by Maven85 on 16/03/2020.
+ */
+
+public class XBMCTextureCache
+{
+  native String _unwrapImageURL(String image);
+
+  private static final String TAG = "@APP_NAME@";
+
+  public XBMCTextureCache()
+  {
+  }
+
+  public String unwrapImageURL(String image)
+  {
+    try
+    {
+      return _unwrapImageURL(image);
+    }
+    catch (Exception e)
+    {
+      Log.e(TAG, "unwrapImageURL: Exception: " + e.getMessage());
+      return null;
+    }
+  }
+
+}

--- a/xbmc/platform/android/activity/CMakeLists.txt
+++ b/xbmc/platform/android/activity/CMakeLists.txt
@@ -16,6 +16,7 @@ set(SOURCES android_main.cpp
             JNIXBMCNsdManagerResolveListener.cpp
             JNIXBMCJsonHandler.cpp
             JNIXBMCFile.cpp
+            JNIXBMCTextureCache.cpp
             JNIXBMCURIUtils.cpp
             JNIXBMCDisplayManagerDisplayListener.cpp
             JNIXBMCSpeechRecognitionListener.cpp
@@ -42,6 +43,7 @@ set(HEADERS AndroidFeatures.h
             JNIXBMCNsdManagerResolveListener.h
             JNIXBMCJsonHandler.h
             JNIXBMCFile.h
+            JNIXBMCTextureCache.h
             JNIXBMCURIUtils.h
             JNIXBMCDisplayManagerDisplayListener.h
             JNIXBMCSpeechRecognitionListener.h

--- a/xbmc/platform/android/activity/JNIXBMCTextureCache.cpp
+++ b/xbmc/platform/android/activity/JNIXBMCTextureCache.cpp
@@ -1,0 +1,42 @@
+/*
+ *  Copyright (C) 2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCTextureCache.h"
+
+#include "CompileInfo.h"
+#include "TextureCache.h"
+
+#include <androidjni/Context.h>
+#include <androidjni/jutils-details.hpp>
+
+using namespace jni;
+
+static std::string s_className = std::string(CCompileInfo::GetClass()) + "/XBMCTextureCache";
+
+void CJNIXBMCTextureCache::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
+  if (cClass)
+  {
+    JNINativeMethod methods[] = {
+        {"_unwrapImageURL", "(Ljava/lang/String;)Ljava/lang/String;",
+         (void*)&CJNIXBMCTextureCache::_unwrapImageURL},
+    };
+
+    env->RegisterNatives(cClass, methods, sizeof(methods) / sizeof(methods[0]));
+  }
+}
+
+jstring CJNIXBMCTextureCache::_unwrapImageURL(JNIEnv* env, jobject thiz, jstring image)
+{
+  std::string strImage = jcast<std::string>(jhstring::fromJNI(image));
+  std::string responseData = CTextureUtils::UnwrapImageURL(strImage);
+
+  jstring jres = env->NewStringUTF(responseData.c_str());
+  return jres;
+}

--- a/xbmc/platform/android/activity/JNIXBMCTextureCache.h
+++ b/xbmc/platform/android/activity/JNIXBMCTextureCache.h
@@ -1,0 +1,29 @@
+/*
+ *  Copyright (C) 2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <androidjni/JNIBase.h>
+
+namespace jni
+{
+
+class CJNIXBMCTextureCache : public CJNIBase
+{
+public:
+  CJNIXBMCTextureCache(const jni::jhobject& object) : CJNIBase(object) {}
+
+  static void RegisterNatives(JNIEnv* env);
+
+protected:
+  ~CJNIXBMCTextureCache() override = default;
+
+  static jstring _unwrapImageURL(JNIEnv* env, jobject thiz, jstring image);
+};
+
+} // namespace jni

--- a/xbmc/platform/android/activity/android_main.cpp
+++ b/xbmc/platform/android/activity/android_main.cpp
@@ -23,6 +23,7 @@
 #include "platform/android/activity/JNIXBMCNsdManagerResolveListener.h"
 #include "platform/android/activity/JNIXBMCSpeechRecognitionListener.h"
 #include "platform/android/activity/JNIXBMCSurfaceTextureOnFrameAvailableListener.h"
+#include "platform/android/activity/JNIXBMCTextureCache.h"
 #include "platform/android/activity/JNIXBMCURIUtils.h"
 #include "platform/android/activity/JNIXBMCVideoView.h"
 
@@ -127,6 +128,7 @@ extern "C" JNIEXPORT jint JNI_OnLoad(JavaVM *vm, void *reserved)
   CJNIXBMCNsdManagerResolveListener::RegisterNatives(env);
   CJNIXBMCSpeechRecognitionListener::RegisterNatives(env);
   CJNIXBMCSurfaceTextureOnFrameAvailableListener::RegisterNatives(env);
+  CJNIXBMCTextureCache::RegisterNatives(env);
   CJNIXBMCURIUtils::RegisterNatives(env);
   CJNIXBMCVideoView::RegisterNatives(env);
 

